apollo-backend/src/test/java/io/logz/apollo/DeploymentTest.java

```diff
@@ -225,4 +229,29 @@ public void testDeploymentToInactiveEnvironment() throws Exception {
         throwableAssert.isInstanceOf(Exception.class);
         throwableAssert.hasMessage("Cannot deploy. Target environment is not active.");
     }
+
+    @Test
+    public void testShouldDeploymentBeCanceled() throws Exception {
+        DeploymentDao deploymentDao = standaloneApollo.getInstance(DeploymentDao.class);
+
+        Deployment deployment = ModelsGenerator.createAndSubmitDeployment(apolloTestClient);
+        int id = deployment.getId();
+        //Canceling a deployment that didn't passed the timeout
+        assertThatThrownBy(() -> apolloTestClient.cancelDeployment(deploymentDao.getDeployment(id)))
```

Comment:
this check is very time sensitive
should request take a little longer or thread stuck for a moment it will fail
it might be better to make two separate tests: negative test (with long `forceCancelTimeoutSeconds`) and `testDeploymentShouldBeCanceled` (same as you did here).
WDYT?

Reply:
but it's the same configuration for all tests, I can make forceCancelTimeoutSeconds a bit longer, but if the request take a little longer or thread stuck for a moment then one if the tests will fail anyway

Reply:
@mayhuberman it's hard to estimate the probability of such thing happening. 
Don't invest too much time if there is no easy way to separate the tests.